flood_adapt/dbs_classes/dbs_template.py

```diff
@@ -42,35 +34,38 @@ def get(self, name: str) -> ObjectModel:
             object of the type of the specified object model
         """
         # Make the full path to the object
-        full_path = self._path / name / f"{name}.toml"
+        full_path = self.input_path / name / f"{name}.toml"
 
         # Check if the object exists
         if not Path(full_path).is_file():
-            raise ValueError(f"{self._type.capitalize()} '{name}' does not exist.")
+            raise ValueError(
+                f"{self._object_class.display_name}: '{name}' does not exist."
+            )
 
         # Load and return the object
-        object_model = self._object_model_class.load_file(full_path)
-        return object_model
+        return self._object_class.load_file(full_path)
 
-    def list_objects(self):
+    def list_objects(self) -> dict[str, list[Any]]:
         """Return a dictionary with info on the objects that currently exist in the database.
 
         Returns
         -------
-        dict[str, Any]
-            Includes 'name', 'description', 'path' and 'last_modification_date' info, as well as the objects themselves
+        dict[str, list[Any]]
+            A dictionary that contains the keys: `name`, 'path', 'last_modification_date', 'description', 'objects'
+            Each key has a list of the corresponding values, where the index of the values corresponds to the same object.
         """
         # Check if all objects exist
         object_list = self._get_object_list()
-        if not all(Path(path).is_file() for path in object_list["path"]):
+        if any(not Path(path).is_file() for path in object_list["path"]):
+            broken = [
+                path.name for path in object_list["path"] if not Path(path).is_file()
+            ]
```

Comment:
small nitpick, but I think you can better create the broken list initially and simply raise if it's non empty, that way you don't have to go through the loop twice

Reply:
I added the check in _get_object_list(), but forgot to remove it here, thanks